infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -784,13 +842,32 @@ private void handleAllocation(RequestMsg req, ChannelHandlerContext ctx, IServer
         Token newToken = new Token(sequencerEpoch, globalLogTail);
         globalLogTail = newTail;
 
+        // handle delete stream
+        if (!txResolutionInfo.getDeleteSet().isEmpty()) {
+            deleteStreams(txResolutionInfo.getDeleteSet());
+        }
+
         // Note: we reuse the request header as the ignore_cluster_id and
         // ignore_epoch fields are the same in both cases.
         ResponseMsg response = getResponseMsg(
                 getHeaderMsg(req.getHeader()), getTokenResponseMsg(newToken, backPointerMap.build()));
         r.sendResponse(response, ctx);
     }
 
+    private Set<UUID> getIllegalStreamsFromRequest(TxResolutionInfo txResolutionInfo) {
```

Comment:
What does it mean for a stream to be illegal?

Reply:
Added a comment in the code. A stream in transaction createSet is considered illegal if the transaction attempts to create it while Sequencer has already allocated token for it. This is used to detect the 'reopening a trimmed table' scenario.